src/workprover/gpusolver.py

```diff
@@ -0,0 +1,106 @@
+import hashlib
+import os.path
+import struct
+
+import utils
+
+pyopencl = None
+numpy = None
+
+class GPUSolverError(Exception):
+    pass
+
+class GPUSolver(object):
+    def __init__(self, codePath, vendors = None):
+        global pyopencl, numpy
+
+        try:
+            import numpy
+            import pyopencl
```

Comment:
Could you please comment #1283 and maybe change the import order too if you agree?